cli/tests/test_parallelize_command.rs

```diff
@@ -0,0 +1,561 @@
+    );
+    test_env.jj_cmd_ok(&workspace_path, &["new", "description(3)", "-m", "4"]);
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @  90a65779e2ec 4
+    ◉  9f5b59fa4622 3
+    │ ◉  a01c1fad8506 2a-c
+    ╭─┤
+    │ ◉  1eb902150bb9 a
+    ◉ │  d826910d21fb 2
+    ◉ │  dc0e5d6135ce 1
+    ├─╯
+    ◉  000000000000
+    "###);
+
+    // After this finishes, child-2a will have three parents: "1", "2", and "a".
+    test_env.jj_cmd_ok(
+        &workspace_path,
+        &["parallelize", "description(1)::description(3)"],
+    );
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @      5a0dd49510d1 4
+    ├─┬─╮
+    │ │ ◉  a9334ecaa379 3
+    │ │ │ ◉  605371712469 2a-c
+    ╭─┬───┤
+    │ │ │ ◉  1eb902150bb9 a
+    │ │ ├─╯
+    │ ◉ │  3a7b37ebe843 2
+    │ ├─╯
+    ◉ │  761e67df44b7 1
+    ├─╯
+    ◉  000000000000
+    "###);
+}
+
+#[test]
+fn test_parallelize_failure_disconnected_target_commits() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let workspace_path = test_env.env_root().join("repo");
+
+    for n in 1..3 {
+        test_env.jj_cmd_ok(&workspace_path, &["commit", &format!("-m{n}")]);
+    }
+    test_env.jj_cmd_ok(&workspace_path, &["describe", "-m=3"]);
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @  9f5b59fa4622 3
+    ◉  d826910d21fb 2
+    ◉  dc0e5d6135ce 1
+    ◉  000000000000
+    "###);
+
+    insta::assert_snapshot!(test_env.jj_cmd_failure(
+        &workspace_path, &["parallelize", "description(1)", "description(3)"]),@r###"
+    Error: Cannot parallelize since the target revisions are not connected.
+    "###);
+}
+
+#[test]
+fn test_parallelize_head_is_a_merge() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let workspace_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_ok(&workspace_path, &["commit", "-m=1"]);
+    test_env.jj_cmd_ok(&workspace_path, &["commit", "-m=2"]);
+    test_env.jj_cmd_ok(&workspace_path, &["new", "root()"]);
+    test_env.jj_cmd_ok(&workspace_path, &["commit", "-m=a"]);
+    test_env.jj_cmd_ok(&workspace_path, &["commit", "-m=b"]);
+    test_env.jj_cmd_ok(
+        &workspace_path,
+        &["new", "description(2)", "description(b)", "-m=merged-head"],
+    );
+    insta::assert_snapshot!(get_log_output(&test_env, &workspace_path), @r###"
+    @    1a8db14a8cf0 merged-head
+    ├─╮
+    │ ◉  401e43e9461f b
+    │ ◉  66ea2ab19a70 a
+    ◉ │  d826910d21fb 2
+    ◉ │  dc0e5d6135ce 1
+    ├─╯
+    ◉  000000000000
+    "###);
+
+    insta::assert_snapshot!(test_env.jj_cmd_failure(&workspace_path,&["parallelize", "description(1)::"]),
+        @r###"
+    Error: Only the roots of the target revset are allowed to have parents which are not being parallelized.
+    "###);
```

Comment:
I can imagine this succeeding and making only 1 and 2 parallel, so merged-head has three parents. I think that might happen kind of automatically if we later rewrite this once I'm done with the rebase API refactoring.

Reply:
You can already do that today by only parallelizing `1::2`. This case fails because it's trying to do `1::` which includes the "merged-head" commit. Just in case that wasn't clear from the test case.